apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/chat/ChatView.kt

```diff
@@ -1401,6 +1422,29 @@ fun BoxWithConstraintsScope.FloatingButtons(
     onLongClick = { showDropDown.value = true }
   )
 
+
+  val lastVisibleItem by remember {
+    derivedStateOf {
+      if (lastIndexOfVisibleItems >= 0) {
+        val lastVisibleChatItemIndex = chatItems.value.lastIndex - firstVisibleIndex - lastIndexOfVisibleItems
+        chatItems.value[lastVisibleChatItemIndex]
+      } else {
+        null
+      }
+    }
+  }
+
+  val firstVisibleItemScrollOffset by remember {
```

Comment:
such solution makes at least two functions to recompose on every scroll. Have you checked the `price` we pay for that solution? 
To check, put this and try to scroll:

```kotlin

SideEffect {
println("LALAL recompose $firstVisibleItemScrollOffset")
}
```